ocaml/typing/printtyp.ml

```diff
@@ -1281,7 +1257,20 @@ let outcome_label : Types.arg_label -> Outcometree.arg_label = function
   | Optional l -> Optional l
   | Position l -> Position l
 
-let rec tree_of_typexp mode ty =
+let tree_of_mode mode l =
+  match mode with
+  | None -> []
+  (* should never raise *)
+  | Some x -> [List.assoc x l]
+
+let tree_of_modes modes =
+  let diff = Mode.Alloc.Const.diff modes Mode.Alloc.Const.legacy in
+  tree_of_mode diff.locality [Mode.Locality.Const.Local, Omd_local] @
+  tree_of_mode diff.linearity [Mode.Linearity.Const.Once, Omd_once] @
+  tree_of_mode diff.uniqueness [Mode.Uniqueness.Const.Unique, Omd_unique]
+
+(* [alloc_mode] is only used if [ty] is arrow-shaped. *)
```

Comment:
But what is it? The mode expected by the context or something? That is, when I'm calling `tree_of_typexp`, how do I know what to pass as `alloc_mode`?